web/pandas_web.py

```diff
@@ -223,7 +224,23 @@ def home_add_releases(context):
         with open(pathlib.Path(context["target_path"]) / "releases.json", "w") as f:
             json.dump(releases, f, default=datetime.datetime.isoformat)
 
-        for release in releases:
+        # Sort the releases in descending order
+        sorted_releases = sorted(
+            releases,
+            key=lambda release: version.parse(release["tag_name"]),
+            reverse=True,
+        )
```

Comment:
I think everything will be clearer if you use something like `sorted(map(version.parse, releases), reverse=True)` instead .

Then you should be able to use something like `if (release.major, release.minor) == (prev_release.major, prev_release.minor): continue` to remove 1.5.3 if 1.5.4 exists.

What do you think?

Reply:
Yeah I agree that would make the code mode readable. Therefore, I just wrote `parsed_releases = map(lambda r: (version.parse(r["tag_name"]), r), releases)` to first put the `releases` into `(version, release)` pairs and then sorted the new list. The version helps to simplify the redundant work of parsing the version again and putting them into the set. However, I'm not sure what version I should initialize `prev_version` to. Right now, I'm using `prev_version = version.Version("0.0.0")` and that seems to work.